Fix Slack notifications when URL is present

Rudedog9d opened the pull request:
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

- Sets URL to null if the current monitor type doesn't utilize the URL field
    - The URL will be saved and used by notifications later if it is not set to null on save
- Utilizes `getUrl` helper to set URL to `null` in monitorJSON if URL is invalid.

Fixes #4864 

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

`ftp://`, `steam://`, and `asdf://` url schemes all supported by Slack:

![image](https://github.com/louislam/uptime-kuma/assets/15984156/e9a5991a-eadc-47c5-aec2-5b4016280c2f)


URL removed on save of DNS monitor: 

![image](https://github.com/louislam/uptime-kuma/assets/15984156/b7b159fc-9389-4358-ac85-5e31f0d66188)

CommanderStorm reviewed (changes requested):
Your change interacts with every notification provider we have.
This is a bit too risky for my taste.
Could you refactor the slack notification privider to use this function instead of using the url parameter?

https://github.com/louislam/uptime-kuma/blob/dc154437164f6ec1b89b8e9f60e1cecc64709a28/server/notification-providers/notification-provider.js#L22-L50

Rudedog9d commented:
Oh yea that's perfect, thanks for pointing that out! I've updated the provider and removed the validation step. I did leave the change to update the produced JSON object though, let me know if that should be removed.

CommanderStorm commented on server/notification-providers/slack.js:
```diff
@@ -48,7 +48,7 @@ class Slack extends NotificationProvider {
         }
 
-        if (monitorJSON.url) {
+        if (this.extractAdress(monitorJSON)) {
```
the extracted `adress` should be used below instead of `monitorJSON.url`.

(I can't suggest this change because of github restrictions)

```suggestion
        const adress = this.extractAdress(monitorJSON);
        if (adress) {
```

Rudedog9d replied:
My bad :facepalm:

CommanderStorm commented on server/model/monitor.js:
```diff
@@ -107,7 +107,7 @@ class Monitor extends BeanModel {
             parent: this.parent,
             childrenIDs: await Monitor.getAllChildrenIDs(this.id),
-            url: this.url,
+            url: this.getUrl()?.href,
```
lets not change this one.
As stated previously, I have not looked into the other notification providers/monitors and don't think that this issue needs to do this.
I am thinking about notification providers such as webhooks allowing templating or another monitor using this field for somethign which is not considered a url by node.

Without good looking into, this is too risky for me.

```suggestion
            url: this.url,
```

CommanderStorm reviewed (changes requested):
I have left comments about the two areas below

CommanderStorm reviewed (approved):
Thanks for the bugfix! 🎉 

<img width="200px" src="https://media4.giphy.com/media/3oEdva9BUHPIs2SkGk/giphy.gif"/>

> [!NOTE]
> This PR is part of the [`v2.0` merge window](https://github.com/louislam/uptime-kuma/milestone/24) => see #4500 for the bugs that need to be addressed before we can ship this release ^^

<sub>All changes in this PR are small and uncontroversial ⇒ merging with junior maintainer approval</sub>
